build.gradle

```diff
@@ -1,74 +1,70 @@
 plugins {
-	id 'fabric-loom' version '0.4-SNAPSHOT'
+	id 'fabric-loom' version '0.10-SNAPSHOT'
+	id 'maven-publish'
+	id "com.github.johnrengelman.shadow" version "7.0.0"
 }
 
-sourceCompatibility = JavaVersion.VERSION_1_8
-targetCompatibility = JavaVersion.VERSION_1_8
+apply plugin: 'java'
+apply plugin: 'fabric-loom'
+apply plugin: 'maven-publish'
+apply plugin: "com.github.johnrengelman.shadow"
+
+defaultTasks('clean', 'build', 'publishToMavenLocal')
+
+sourceCompatibility = '1.16'
+targetCompatibility = '1.16'
```

Comment:
You should compile for java 17, as that is the latest version.

Additionally, setting it to `1.16` does not compile for java 16. You need to set it to `16`.